pkg/operator/bootstrap.go

```diff
@@ -33,30 +32,35 @@ func resourceName(namespace string) string {
 	return imageregistryv1.ImageRegistryResourceName
 }
 
+func deploymentConfigName(namespace string) string {
+	if namespace == "default" {
+		return "docker-registry"
+	}
+	return imageregistryv1.ImageRegistryName
+}
+
 func (c *Controller) Bootstrap() error {
 	client, err := regopset.NewForConfig(c.kubeconfig)
 	if err != nil {
 		return err
 	}
-	crList, err := c.listers.RegistryConfigs.List(labels.Everything())
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			return fmt.Errorf("failed to list registry custom resources: %s", err)
-		}
+
+	cr, err := c.listers.RegistryConfigs.Get(resourceName(c.params.Deployment.Namespace))
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("unable to get the registry custom resources: %s", err)
 	}
 
-	switch len(crList) {
-	case 0:
-		// let's create it.
-	case 1:
+	// If the registry resource already exists,
+	// no bootstrapping is required
+	if cr != nil {
 		return nil
-	default:
-		return fmt.Errorf("only one registry custom resource expected in %s namespace, got %d", c.params.Deployment.Namespace, len(crList))
 	}
 
+	// If no registry resource exists,
+	// let's create one
 	var spec imageregistryv1.ImageRegistrySpec
 
-	dc, err := c.listers.DeploymentConfigs.Get(resourceName(c.params.Deployment.Namespace))
+	dc, err := c.listers.DeploymentConfigs.Get(deploymentConfigName(c.params.Deployment.Namespace))
```

Comment:
the adoption/migration code is increasingly out of sync and unnecessary, this is probably a good opportunity to remove it.

Reply:
updated in current push